### DISCUSSION TITLE
Traits not getting generated

### DISCUSSION THREAD

**Discussion by yuvyakubov** [nftchef/art-engine#177] [Help + Q&A]
I am trying to generate around 1500 images. I have around 200 traits, I have seen that when generating some of my rare traits #0.2 do not get generated. I have around 10 that don't get generated. Is there a way to ensure that my rare traits get generated at least once and maximum of 7 times?

**Comment by BakeSloth** [comment]
Are you using subfolders?
if so, you might have your math wrong. The rarity of the filename inherits the rarity of the folder.
So if your folder structure is -
-Layer
--Folder#50
---filename#50.png

Then filename.png trait has a 25% chance of coming up, not 50%.
50% of 50% = 25%

I believe you can also use the ultrarare config to force a 1 of 1 type nft to be injected into the collection if you want to have exact control over the rarity of an nft as a whole. might be worth looking into on the read me.
node utils/replace.js ./ultraRares
